src/Engines/AlgoliaEngine.php

```diff
@@ -171,20 +170,16 @@ public function mapIds($results)
     public function map(Builder $builder, $results, $model)
     {
         if (count($results['hits']) === 0) {
-            return Collection::make();
+            return $model->newCollection();
         }
 
-        $models = $model->getScoutModelsByIds(
-            $builder, collect($results['hits'])->pluck('objectID')->values()->all()
-        )->keyBy(function ($model) {
-            return $model->getScoutKey();
-        });
+        $objectIds = collect($results['hits'])->pluck('objectID')->values()->all();
```

Comment:
Your refactorization is a great job, but I would prefer that we keep this modification simple, with **only** the introduction of the `$model->newCollection()`:
```php
    public function map(Builder $builder, $results, $model)
    {
        if (count($results['hits']) === 0) {
            return $model->newCollection();
        }

        $models = $model->getScoutModelsByIds(
            $builder, collect($results['hits'])->pluck('objectID')->values()->all()
        )->keyBy(function ($model) {
            return $model->getScoutKey();
        });

        return $model->newCollection($results['hits'])->map(function ($hit) use ($models) {
            if (isset($models[$hit['objectID']])) {
                return $models[$hit['objectID']];
            }
        })->filter()->values();
    }
```

**Reason**: The current contract specified on the phpdocs says that `getScoutModelsByIds` must return mixed. Some people may return an array out of it.

Reply:
@nunomaduro at the moment there's a `keyBy` call right after the `getScoutModelsByIds` call so there's no way the current implementation can work without a collection object.